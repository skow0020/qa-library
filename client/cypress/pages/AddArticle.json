{
    "title": "#title",
    "author": "#author",
    "category": "#category",
    "language": "#language",
    "url": "#url",
    "backgroundImage": "#backgroundImage",
    "description": "#body"
}