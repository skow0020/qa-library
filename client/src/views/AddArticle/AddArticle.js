/* eslint jsx-a11y/anchor-is-valid: 0 */

import AlertModal, { showAlert } from "../../components/common/AlertModal";
import {
  Button,
  Col,
  Container,
  Form,
  FormInput,
  FormTextarea,
  Row
} from "shards-react";

import CategoriesSelection from "../../components/common/CategoriesSelection";
import LanguagesSelection from "../../components/common/LanguagesSelection";
import PageTitle from "../../components/common/PageTitle";
import React from "react";
import axios from "axios";

class AddArticle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: 0,
      backgroundImage: '',
      author: '',
      category: 'General',
      url: '',
      title: '',
      body: '',
      language: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const article = {
      backgroundImage: this.state.backgroundImage,
      author: this.state.author,
      category: this.state.category,
      url: this.state.url,
      title: this.state.title,
      body: this.state.body,
      language: this.state.language
    };

    axios.post("/api/articles/", article)
      .then(
        response => {
          if (response.data.success) { this.showNotifier("Article added successfully"); }
          else { this.showNotifier(`Unable to add article: ${JSON.stringify(response.data.error.errors)}`); }
          this.nextPath('/articles');
        },
        error => this.showNotifier(`Unable to add article: ${error}`)
      );
  }

  nextPath(path) {
    setTimeout(function () {
      this.props.history.push(path);
    }.bind(this), 2000);
  }

  showNotifier = (message) => {
    showAlert({ message });
  }

  render() {
    return (
      <Container fluid className="main-content-container px-4">
        <AlertModal />
        <Form id="add-article-form" onSubmit={this.handleSubmit}>
          <Row noGutters className="page-header py-4">
            <PageTitle sm="8" title="Add an Article" className="text-sm-left" />
            <Button id="submit" type="submit" className="btn btn-success btn-lg">
              Submit
            </Button>
          </Row>

          <label className="text-muted font-weight-bold" htmlFor="title">Article Title</label>
          <FormInput id="title" type="text" value={this.state.title} onChange={this.handleChange} required />
          <Row form>
            <Col md="4" className="form-group">
              <label className="text-muted font-weight-bold" htmlFor="author">Article Author</label>
              <FormInput id="author" type="text" value={this.state.author} onChange={this.handleChange} required />
            </Col>
            <LanguagesSelection value={this.state.language} onChange={this.handleChange}/>
            <CategoriesSelection value={this.state.category} onChange={this.handleChange} required />
          </Row>
          <label className="text-muted font-weight-bold" htmlFor="url">Article URL</label>
          <FormInput id="url" type="text" value={this.state.url} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="backgroundImage">Background Image URL</label>
          <FormInput id="backgroundImage" type="text" value={this.state.backgroundImage} onChange={this.handleChange} required />
          <label className="text-muted font-weight-bold" htmlFor="body">Article Description</label>
          <FormTextarea id="body" type="textarea" style={{ height: "150px" }} value={this.state.body} onChange={this.handleChange} />
        </Form>
      </Container>
    );
  }
}

export default AddArticle;
