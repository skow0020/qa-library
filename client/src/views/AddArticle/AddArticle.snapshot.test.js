import AddArticle from './AddArticle';
import React from 'react';
import renderer from 'react-test-renderer';

describe('AddArticle Snapshot Tests', () => {
  test('AddArticle snapshot', () => {
    const component = renderer.create(
      <AddArticle></AddArticle>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});