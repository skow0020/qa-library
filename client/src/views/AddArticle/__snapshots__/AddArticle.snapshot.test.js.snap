// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArticle Snapshot Tests AddArticle snapshot 1`] = `
<div
  className="main-content-container px-4 container-fluid"
>
  <div />
  <form
    className=""
    id="add-article-form"
    onSubmit={[Function]}
  >
    <div
      className="page-header py-4 no-gutters row"
    >
      <div
        className="text-sm-left text-center text-md-left mb-sm-0 col-12 col-sm-8"
      >
        <span
          className="text-uppercase page-subtitle"
        />
        <h3
          className="page-title"
        >
          Add an Article
        </h3>
      </div>
      <button
        className="btn btn-success btn-lg btn btn-primary"
        id="submit"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="title"
    >
      Article Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <div
      className="form-row"
    >
      <div
        className="form-group col-md-4"
      >
        <label
          className="text-muted font-weight-bold"
          htmlFor="author"
        >
          Article Author
        </label>
        <input
          className="form-control"
          id="author"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="col"
      >
        <label
          className="text-muted font-weight-bold px-2"
          htmlFor="language"
        >
          Languages
        </label>
        <select
          className="form-control custom-select"
          id="language"
          onChange={[Function]}
          type="text"
          value=""
        >
          <option />
          <option>
            Java
          </option>
          <option>
            CSharp
          </option>
          <option>
            Python
          </option>
          <option>
            Ruby
          </option>
          <option>
            JavaScript
          </option>
          <option>
            Cpp
          </option>
          <option>
            Swift
          </option>
        </select>
      </div>
      <div
        className="col"
      >
        <label
          className="text-muted font-weight-bold px-2"
          htmlFor="category"
        >
          Categories
        </label>
        <select
          className="form-control custom-select"
          id="category"
          onChange={[Function]}
          required={true}
          type="text"
          value="General"
        >
          <option />
          <option>
            General
          </option>
          <option>
            UI Automation
          </option>
          <option>
            API Automation
          </option>
          <option>
            Unit Testing
          </option>
          <option>
            DevOps
          </option>
          <option>
            Development
          </option>
          <option>
            Databases
          </option>
        </select>
      </div>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="url"
    >
      Article URL
    </label>
    <input
      className="form-control"
      id="url"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="backgroundImage"
    >
      Background Image URL
    </label>
    <input
      className="form-control"
      id="backgroundImage"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="body"
    >
      Article Description
    </label>
    <textarea
      className="form-control"
      id="body"
      onChange={[Function]}
      style={
        Object {
          "height": "150px",
        }
      }
      type="textarea"
      value=""
    />
  </form>
</div>
`;
